File: 144. Binary Tree Preorder Traversal/recursive.cpp
class Solution {
public:
    vector<int> preorderTraversal(TreeNode* root) {
        vector<int> res;
        if (root != NULL) {
            res.push_back(root->val);

            vector<int> left = preorderTraversal(root->left);
            res.insert(res.end(), left.begin(), left.end());

            vector<int> right = preorderTraversal(root->right);
            res.insert(res.end(), right.begin(), right.end()); 
        }   

        return res;
    }
};

File: 98. Validate Binary Search Tree/recursive.py
class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        return self.recursive(root, float('inf'), float('-inf'))
        
    def recursive(self, root, upper, lower):
        if root is None:
            return True
            
        if root.val >= upper or root.val <= lower:
            return False
            
        return self.recursive(root.left, root.val, lower) and \
            self.recursive(root.right, upper, root.val)